### ISSUE TITLE
Invalid Bundle, asset valiation failure on upload

### ISSUE THREAD

**Issue by nicolopadovan** [mapbox/mapbox-search-ios#191]
## Environment

- Xcode version: 15.3
- iOS version: 15.0+
- Devices affected: *
- Maps SDK Version: -

Packages versions:
- MapboxCommon: 23.6.0
- MapboxCoreMaps: 10.14.0
- MapboxDirections: 2.12.0
- MapboxMaps: 10.14.0
- MapboxMobileEvents: 1.0.10
- MapboxNavigation: 2.14.0
- MapboxNavigationNative: 137.1.7
- MapboxSearch: 1.0.0-rc.7
- MapboxSpech: 2.1.1

## Observed behavior and steps to reproduce

The application builds, runs and archives correctly. However, when uploading (either to App Store or TestFlight only), multiple errors related to the `MapboxCoreSearch` package Info.plist file are yielded.
This only happens after the last Xcode update.
In my specific case, I've installed Mapbox via Swift Package Manager.

The errors that are yielded are:
```
Asset validation failed
The bundle 'Payload/[MyApp].app/Frameworks/MapboxCoreSearch.framework' is missing plist key. 
The Info.plist file is missing the required key: CFBundleShortVersionString. 
Please find more information about CFBundleShortVersionString at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring
```

```
Asset validation failed
This bundle Payload/[MyApp].app/Frameworks/MapboxCoreSearch.framework is invalid.
The Info.plist file is missing the required key: CFBundleVersion.
Please find more information about CFBundleVersion at https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion
```

```
Asset validation failed
Invalid Bundle.
The bundle [MyApp].app/Frameworks/MapboxCoreSearch.framework does not support the minimum OS Version specified in the Info.plist.
```

## Expected behavior

The application should upload normally.

## Notes / preliminary analysis

None of the following works:
- Updating the package to the latest version
- Resolving package version
- Resetting cache
- Cleaning derived data

## Additional links and references

It seems that the Firebase iOS SDK encountered the same issue a couple of weeks ago. [Link](https://github.com/firebase/firebase-ios-sdk/issues/12441)

**Comment by aokj4ck** [comment]
@nicolopadovan thank you for this bug report. I am actively investigating the issue.

**Comment by nicolopadovan** [comment]
> @nicolopadovan thank you for this bug report. I am actively investigating the issue.

Thank you for investigating. Is there any update on the issue?

**Comment by nicolopadovan** [comment]
> @aokj4ck We are seeing the same issue with Mapbox Search 2.0.0-rc.3. This is a serious issue that is blocking the release of our app. If this is not resolved we are going to have to drop Mapbox Search from our app, so a prompt response would be appreciated. <img alt="Screenshot 2024-04-04 at 10 02 28" width="1143" src="https://private-user-images.githubusercontent.com/23720317/319497717-96513529-b664-49c1-ad65-f620ab78228a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMTg0NDcsIm5iZiI6MTcxMjIxODE0NywicGF0aCI6Ii8yMzcyMDMxNy8zMTk0OTc3MTctOTY1MTM1MjktYjY2NC00OWMxLWFkNjUtZjYyMGFiNzgyMjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDA4MDkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyZDhmMGQyYzQ4MGZlNjFjMmFmZDFiYzY5NDY4NTUyZTA5ZGZhODkxNWNkNjc2NTZiNjRkZmQ4MTk1ZWM0OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VNRmYQEtO8q4EzraumYlHwky2_ZsihQ9RJ-UOaQc724">

I’m facing the same issue. For now, I had to revert to the previous Xcode version, downloading it from the developer portal. 
You may encounter some issues if you already updated the packages for the new Xcode version, and you may need to re-install them.
Currently from my phone, but I’ll link the precise Xcode version to use asap.

**Comment by MarnixArnold** [comment]
>I had to revert to the previous Xcode version
Thanks for the tip @nicolopadovan, we did manage to upload to Test Flight with Xcode 15.2. Hardly a permanent solution of course, but this gives us a bit of breathing room at least...

**Comment by nicolopadovan** [comment]
Is there any update on the issue yet? @aokj4ck

**Comment by MarnixArnold** [comment]
I've tried updating to MapboxSearch 1.0.0-rc9 but the problem persists. @aokj4ck is this going to be fixed any time soon? Or at all? We would like to be able to use the latest Xcode again...

**Comment by aokj4ck** [comment]
@MarnixArnold @nicolopadovan I'm preparing a new release to resolve this. In the meantime until that release is finalized you can use this commit https://github.com/mapbox/mapbox-search-ios/commit/e862c84c3c1e6a186d6c243403daf811f3dbcece with SPM to resolve this issue with the v1 version series. Please let me know if you encounter any other issues. I'll update this issue once that release is published.

**Comment by MarnixArnold** [comment]
@aokj4ck we've tried release 1.0.1 with Xcode 15.3 and while the first two errors (`CFBundleShortVersionString` and `CFBundleVersion`) have gone, the third is still there:
```
Asset validation failed, NSLocalizedRecoverySuggestion=Invalid Bundle. The bundle MyApp.app/Frameworks/MapboxCoreSearch.framework does not support the minimum OS Version specified in the Info.plist
```

This is because your framework's min OS version is 16.5 but its Info.plist specifies 11.0:
```
vtool -show MapboxCoreSearch.framework/MapboxCoreSearch 
MapboxCoreSearch.framework/MapboxCoreSearch:
Load command 8
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform IOS
    minos 16.5
      sdk 17.4
   ntools 1
     tool LD
  version 1053.12
Load command 9
      cmd LC_SOURCE_VERSION
  cmdsize 16
  version 0.0
```
Info.plist:
```
	<key>MinimumOSVersion</key>
	<string>11.0</string>
```

Maybe you can do like Google did for the FirebaseAnalytics and just set it to 100.0, that worked for them:
```
	<key>MinimumOSVersion</key>
	<string>100.0</string>
```

**Comment by aokj4ck** [comment]
@MarnixArnold yes this is a known issue with Xcode 15.3 and in the meantime you might have luck with Xcode 15.2 instead. I am working to fix this for you and appreciate the note. Hm `vtool` reporting minos 16.5 is not what I anticipated. Are you using Swift Package Manager, CocoaPods, or another package manager (like direct downloads or Carthage)?

**Comment by MarnixArnold** [comment]
@aokj4ck Yes, so far we've been able to make builds using Xcode 15.2 but with 15.4 around the corner there's no telling how long Apple will continue to allow us to do that.

We are using SPM only.

**Comment by MarnixArnold** [comment]
@aokj4ck I've just made a build with MapboxSearch 1.0.2 on Xcode 15.3 and it validates, so it looks like the problems have been resolved.

**Comment by aokj4ck** [comment]
@MarnixArnold please update to our latest release v2.0.3 and let me know if you encounter any issues. https://github.com/mapbox/mapbox-search-ios/releases/tag/v2.0.3

**Comment by aokj4ck** [comment]
@nicolopadovan is this issue solved for you with [release v1.0.2](https://github.com/mapbox/mapbox-search-ios/releases/tag/v1.0.2)?
